### PR TITLE
Navigation: Improve trigger for fallback navigation

### PR DESCRIPTION
**Pull request by tyxla**
## What?
Come up with an alternative solution for solving the problem #50321 intends to solve.

## Why?
#50321 uses an external flag for the component, which is against React's rules, and the React Compiler doesn't appreciate it (see #61788).

## How?
The goal of #50321 was to create fallback navigation only if there were no navigations at all, and create it only once for the session. 

This PR achieves it by checking if the request to create a fallback navigation is currently resolving. This simplifies the logic by avoiding an extra flag, and satisfies the React Compiler. 

## Testing Instructions
Same as in #50321.

### Testing Instructions for Keyboard
Same

## Screenshots or screencast 
None

### CODE REVIEW

File: packages/edit-site/src/components/sidebar-navigation-screen-navigation-menus/index.js [hunk 1]
```diff
@@ -61,21 +58,22 @@ export default function SidebarNavigationScreenNavigationMenus( { backPath } ) {
...
 	// then trigger fallback algorithm to create one.
 	if (
 		! firstNavigationMenu &&
 		! isResolvingNavigationMenus &&
 		hasResolvedNavigationMenus &&
-		! hasCreatedFallback
+		// Ensure a fallback navigation is created only once
+		! isCreatingNavigationFallback
 	) {
 		getNavigationFallbackId();
```
Comment (Mamaduka) [inline packages/edit-site/src/components/sidebar-navigation-screen-navigation-menus/index.js:78]: Calling a selector during the render seems odd. I know that useSelect isn't the best place to "fire and forget" pattern, but I can't think of a better alternative.

**Comment by draganescu** [comment]
Juts for history: the flag removed in this PR was added in #51413 not in #50321 in [this commit](https://github.com/WordPress/gutenberg/pull/51413/commits/06ad9ce4f7b65cf3db69b4a618820dfa8de09231).

File: packages/edit-site/src/components/sidebar-navigation-screen-navigation-menus/index.js (hunk 1 above)
Comment (tyxla) [reply packages/edit-site/src/components/sidebar-navigation-screen-navigation-menus/index.js:78]: I admit that's weird, however, with the mechanics of selectors and resolvers, it does make sense to some degree. It's also documented why we're doing it, so it felt OK to leave as-is. This PR attempts to only fix the React Compiler error after all.

**Review by draganescu — APPROVED** [review]
I've tested this and it seems to be OK. I remember back in the day these navigation block integrations were added sometimes checking for `isResolving` was insufficient 🤷🏻 

All in all 🚀 let's bring this in. 🤞🏻

**Pull request merged**